components/admin/data/layers/form/LayersForm.js

```diff
@@ -63,22 +64,18 @@ class LayersForm extends PureComponent {
       authorization: props.authorization,
       language: props.locale
     });
-    this.datasetsService = new DatasetsService({
-      authorization: props.authorization,
-      language: props.locale
-    });
 
     this.layerManager = new LayerManager(null, {
       layersUpdated: (valid, err) => {
-        this.props.dispatch(setLayerInteractionError(valid ? false : err));
+        this.props.setLayerInteractionError(valid ? false : err);
       }
     });
   }
 
   componentDidMount() {
     const { id } = this.state;
 
-    const promises = [this.datasetsService.fetchAllData({})];
+    const promises = [fetchDatasets({ 'page[size]': 999999 })];
```

Comment:
One day will need to change this kind of behavior for fetching by demand or similar... It's kinda crazy to bring everything at once... Just pointing this 👍